provider/pkg/provider/provider.go

```diff
@@ -2955,6 +2955,11 @@ func checkpointObject(inputs, live *unstructured.Unstructured, fromInputs resour
 		}
 	}
 
+	inputsCopy := resource.NewObjectProperty(inputsPM)
```

Comment:
Hey @lblackstone I'm realizing your comment isn't quite enough for me to make changes.

The empty input map is only one issue. Another is that `ContainsSecret()` is false here, so that would require further changes to `annotateSecrets`. We need to check `GetKind()` but `annotateSecrets` works with property maps that don't even have this method.

As for `data` and `stringdata`, I don't see what needs to be changed there.

Reply:
Synced offline - latest revision should reflect all the input. Thank you!